Validation tests fail due to possible race condition

**Describe the issue**
Three test_validation.py tests fail - primarily due to an assertion failure against the length of the issues list.  These do not occur when the test is run by itself, but if the test_handlers.py tests are run prior (within the same pytest invocation), then the len(issues) assertion is triggered. 

**To Reproduce**
Steps to reproduce the behavior:
1. Run `pytest elyra/pipeline/tests/test_handlers.py elyra/pipeline/tests/test_validation.py` from the elyra repo root.


**Screenshots or log output**
```
FAILED elyra/pipeline/tests/test_validation.py::test_invalid_runtime_node_kubeflow - assert 2 == 1
FAILED elyra/pipeline/tests/test_validation.py::test_invalid_runtime_node_kubeflow_with_supernode - assert 3 == 1
FAILED elyra/pipeline/tests/test_validation.py::test_missing_node_property_for_kubeflow_pipeline - KeyError: 'current_parameters'
```

**Expected behavior**
Tests should succeed and pass

**Deployment information**
Describe what you've deployed and how:
 - Elyra version: master
 - Installation source: source
 - Deployment type: local
 - Operating system: macos

--- a/elyra/pipeline/tests/conftest.py
+++ b/elyra/pipeline/tests/conftest.py
@@ -1,0 +1,40 @@
+#
+# Copyright 2018-2021 Elyra Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import shutil
+
+import pytest
+
+
+@pytest.fixture
+def setup_factory_data(jp_environ, jp_env_jupyter_path):
+    """Copies the factory metadata instances for runtime-images and compontent-registries to test hierarchy."""
+    source = os.path.join(os.path.dirname(__file__), '../../../etc/config/metadata')
+    destination = os.path.join(jp_env_jupyter_path, 'metadata')
+    shutil.copytree(source, destination)
+    yield destination  # this return value probably won't be used, but here nonetheless
+
+
+# Set Elyra server extension as enabled (overriding server_config fixture from jupyter_server)
+@pytest.fixture
+def jp_server_config(setup_factory_data):
+    return {
+        "ServerApp": {
+            "jpserver_extensions": {
+                "elyra": True
+            }
+        }
+    }

--- a/elyra/pipeline/tests/test_handlers.py
+++ b/elyra/pipeline/tests/test_handlers.py
@@ -16,8 +16,6 @@
 
 import json
 
-import pytest
-
 from elyra.pipeline.tests import resources
 
 # from jupyter_server.tests.utils import expected_http_error
@@ -28,18 +26,6 @@ try:
 except ImportError:
     # Try backported to PY<37 `importlib_resources`.
     import importlib_resources as pkg_resources
-
-
-# Set Elyra server extension as enabled (overriding server_config fixture from jupyter_server)
-@pytest.fixture
-def jp_server_config():
-    return {
-        "ServerApp": {
-            "jpserver_extensions": {
-                "elyra": True
-            }
-        }
-    }
 
 
 async def test_get_components(jp_fetch):

--- a/elyra/pipeline/tests/test_processor_airflow.py
+++ b/elyra/pipeline/tests/test_processor_airflow.py
@@ -32,7 +32,7 @@ PIPELINE_FILE = 'resources/sample_pipelines/pipeline_dependency_complex.json'
 
 
 @pytest.fixture
-def processor():
+def processor(setup_factory_data):
     processor = AirflowPipelineProcessor(os.getcwd())
     return processor
 

--- a/elyra/pipeline/tests/test_processor_kfp.py
+++ b/elyra/pipeline/tests/test_processor_kfp.py
@@ -33,7 +33,7 @@ from elyra.pipeline.tests.test_pipeline_parser import _read_pipeline_resource
 
 
 @pytest.fixture
-def processor():
+def processor(setup_factory_data):
     processor = KfpPipelineProcessor(os.getcwd())
     return processor
 

--- a/elyra/pipeline/tests/test_validation.py
+++ b/elyra/pipeline/tests/test_validation.py
@@ -37,7 +37,7 @@ def load_pipeline():
 
 
 @pytest.fixture
-def validation_manager():
+def validation_manager(setup_factory_data):
     root = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__), "resources/validation_pipelines"))
     yield PipelineValidationManager.instance(root_dir=root)
     PipelineValidationManager.clear_instance()

